xcm/src/v1/multilocation.rs

```diff
@@ -476,254 +818,301 @@ impl MultiLocation {
+				parents: 1,
+				interior: X6(
+					j1.try_into()?,
+					j2.try_into()?,
+					j3.try_into()?,
+					j4.try_into()?,
+					j5.try_into()?,
+					j6.try_into()?,
+				),
+			}),
+			MultiLocation0::X7(j0, j1, j2, j3, j4, j5, j6) => Ok(X7(
+				j0.try_into()?,
+				j1.try_into()?,
+				j2.try_into()?,
+				j3.try_into()?,
+				j4.try_into()?,
+				j5.try_into()?,
+				j6.try_into()?,
+			)
+			.into()),
+			MultiLocation0::X8(j0, j1, j2, j3, j4, j5, j6, j7)
+				if j0.is_parent() &&
+					j1.is_parent() && j2.is_parent() &&
+					j3.is_parent() && j4.is_parent() &&
+					j5.is_parent() && j6.is_parent() &&
+					j7.is_parent() =>
+				MultiLocation::with_parents(8),
+			MultiLocation0::X8(j0, j1, j2, j3, j4, j5, j6, j7)
+				if j0.is_parent() &&
+					j1.is_parent() && j2.is_parent() &&
+					j3.is_parent() && j4.is_parent() &&
+					j5.is_parent() && j6.is_parent() =>
+				Ok(MultiLocation { parents: 7, interior: X1(j7.try_into()?) }),
+			MultiLocation0::X8(j0, j1, j2, j3, j4, j5, j6, j7)
+				if j0.is_parent() &&
+					j1.is_parent() && j2.is_parent() &&
+					j3.is_parent() && j4.is_parent() &&
+					j5.is_parent() =>
+				Ok(MultiLocation { parents: 6, interior: X2(j6.try_into()?, j7.try_into()?) }),
+			MultiLocation0::X8(j0, j1, j2, j3, j4, j5, j6, j7)
+				if j0.is_parent() &&
+					j1.is_parent() && j2.is_parent() &&
+					j3.is_parent() && j4.is_parent() =>
+				Ok(MultiLocation {
+					parents: 5,
+					interior: X3(j5.try_into()?, j6.try_into()?, j7.try_into()?),
+				}),
+			MultiLocation0::X8(j0, j1, j2, j3, j4, j5, j6, j7)
+				if j0.is_parent() && j1.is_parent() && j2.is_parent() && j3.is_parent() =>
+				Ok(MultiLocation {
+					parents: 4,
+					interior: X4(j4.try_into()?, j5.try_into()?, j6.try_into()?, j7.try_into()?),
+				}),
+			MultiLocation0::X8(j0, j1, j2, j3, j4, j5, j6, j7)
+				if j0.is_parent() && j1.is_parent() && j2.is_parent() =>
+				Ok(MultiLocation {
+					parents: 3,
+					interior: X5(
+						j3.try_into()?,
+						j4.try_into()?,
+						j5.try_into()?,
+						j6.try_into()?,
+						j7.try_into()?,
+					),
+				}),
+			MultiLocation0::X8(j0, j1, j2, j3, j4, j5, j6, j7)
+				if j0.is_parent() && j1.is_parent() =>
+				Ok(MultiLocation {
+					parents: 2,
+					interior: X6(
+						j2.try_into()?,
+						j3.try_into()?,
+						j4.try_into()?,
+						j5.try_into()?,
+						j6.try_into()?,
+						j7.try_into()?,
+					),
+				}),
+			MultiLocation0::X8(j0, j1, j2, j3, j4, j5, j6, j7) if j0.is_parent() =>
+				Ok(MultiLocation {
+					parents: 1,
+					interior: X7(
+						j1.try_into()?,
+						j2.try_into()?,
+						j3.try_into()?,
+						j4.try_into()?,
+						j5.try_into()?,
+						j6.try_into()?,
+						j7.try_into()?,
+					),
+				}),
+			MultiLocation0::X8(j0, j1, j2, j3, j4, j5, j6, j7) => Ok(X8(
+				j0.try_into()?,
+				j1.try_into()?,
+				j2.try_into()?,
+				j3.try_into()?,
+				j4.try_into()?,
+				j5.try_into()?,
+				j6.try_into()?,
+				j7.try_into()?,
+			)
+			.into()),
```

Comment:
I read all of this impl to double check, I think a macro to generate the match is easier to verify, as a future enhancement.